src/App.tsx

```diff
@@ -15,48 +17,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export const App: FC = () => {
+  const getReorderedGoods = (
+    goods: string[],
+    { sortType, isReversed }: ReorderOptions,
+  ) => {
+    const visibleGoods = [...goods];
+
+    visibleGoods.sort((firstItem, secondItem) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return firstItem.localeCompare(secondItem);
+
+        case SortType.LENGTH:
+          return firstItem.length - secondItem.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
```

Comment:
Let's remove this function outside your component `app` to avoid recreating every re-render